src/rpc/net.cpp

```diff
@@ -415,7 +415,16 @@ static RPCHelpMan disconnectnode()
 
     if (!address_arg.isNull() && id_arg.isNull()) {
         /* handle disconnect-by-address */
-        success = connman.DisconnectNode(address_arg.get_str());
+        if (address_arg.get_str().find('/') != std::string::npos) {
+            CSubNet subnet;
+            if (LookupSubNet(address_arg.get_str(), subnet)) {
+                success = connman.DisconnectNode(subnet);
+            } else {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid subnet");
+            }
+        } else {
+            success = connman.DisconnectNode(address_arg.get_str());
+        }
```

Comment:
Why not drop this entirely? Seems like a bug specifying an IP won't disconnect all nodes connecting from that IP...

Reply:
In fact, maybe this should be split into a bugfix commit before adding subnet support.

Reply:
I don't think it's really a bug. It seems like this was the expected behavior (due to examples and docs - e.g. having to specify port). However, I also think it should be able to disconnect all nodes from an IP. I'm gonna address it.